lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
Aw man, I realized that yesterday and forgot by the time I posted.

I think that forces us to use:

```js
// multiple
topic.publish([
  { data: "anyType", labels: [] },
  { data: "anyType", labels: [] }
], callback)

// single
topic.publish({ data: "anyType", labels: [] }, callback)
```

We could use a message type?

```js
var message = topic.message("Hi!")
message.labels = ["user", "group3"]
// message.addLabel("user")
// message.addLabel("group3")

topic.publish(message, callback)
```